net-misc/unison/unison-2.51.4_rc2.ebuild

```diff
@@ -0,0 +1,107 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit desktop xdg-utils
+
+DESCRIPTION="Two-way cross-platform file synchronizer"
+HOMEPAGE="https://www.seas.upenn.edu/~bcpierce/unison/"
+SRC_URI="https://github.com/bcpierce00/unison/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="$(ver_cut 1-2)"
+KEYWORDS="~amd64 ~arm ~ppc ~x86 ~amd64-linux ~x86-linux ~ppc-macos ~sparc-solaris"
+IUSE="debug doc gtk +ocamlopt threads"
+RESTRICT="!ocamlopt? ( strip )" # https://bugs.gentoo.org/685776#c0
+
+# ocaml version so we are sure it has ocamlopt use flag
+BDEPEND="dev-lang/ocaml:=[ocamlopt?]
+	doc? ( app-text/dvipsk
+		app-text/ghostscript-gpl
+		dev-texlive/texlive-latex )"
+DEPEND="gtk? ( dev-ml/lablgtk:2= )"
+RDEPEND="gtk? ( dev-ml/lablgtk:2=
+	|| ( net-misc/x11-ssh-askpass net-misc/ssh-askpass-fullscreen ) )
+	>=app-eselect/eselect-unison-0.4"
+
+DOCS=( BUGS.txt CONTRIB INSTALL NEWS README ROADMAP.txt TODO.txt )
+
+src_prepare() {
+	# https://github.com/bcpierce00/unison/issues/416
+	sed -e "/ifdef\ HEVEA/,/endif/d" -i doc/Makefile || die
+	# https://github.com/bcpierce00/unison/pull/415
+	sed -e "/myName/d" -i doc/docs.ml || die
+	default
+}
+
+src_compile() {
+	local myconf
+
+	if use threads; then
+		myconf="$myconf THREADS=true"
+	fi
+
+	if use debug; then
+		myconf="$myconf DEBUGGING=true"
+	fi
+
+	if use gtk; then
+		myconf="$myconf UISTYLE=gtk2"
+	else
+		myconf="$myconf UISTYLE=text"
+	fi
+
+	use ocamlopt || myconf="$myconf NATIVE=false"
+
+	if use doc; then
+		VARTEXFONTS="${T}/fonts" emake $myconf CFLAGS="" HEVEA=false docs
+	fi
+
+	# Discard cflags as it will try to pass them to ocamlc...
+	emake $myconf CFLAGS="" src
```

Comment:
I'd say you could use a single-liner with `usex` to toggle these, but it could become a bit messy. Please at least update it to an array form, 
```
	local myconf=()

	if use threads; then
		myconf+=( THREADS=true )
	fi

	...
	...
	
	emake "${myconf[@]]}" CFLAGS="" src
```

Reply:
Done